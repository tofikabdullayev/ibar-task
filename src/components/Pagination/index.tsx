import React from 'react';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';

interface Props {
  pageCount: number;
  currentPage: number;
  selectPage: (pageNumber: number) => void;
}

const PaginationComponent = ({ pageCount, currentPage, selectPage }: Props) => {
  const pages = [...Array(pageCount)].map((e, i) => (
    <PaginationItem active={currentPage === i + 1} key={`${i}_${Date.now()}`}>
      <PaginationLink onClick={() => selectPage(i + 1)}>{i + 1}</PaginationLink>
    </PaginationItem>
  ));

  return (
    <Pagination
      aria-label="Page navigation"
      style={{ marginTop: 20, flexWrap: 'wrap' }}
    >
      <PaginationItem disabled={currentPage === 1}>
        <PaginationLink previous onClick={() => selectPage(currentPage - 1)} />
      </PaginationItem>

      {pages}

      <PaginationItem disabled={currentPage === pageCount}>
        <PaginationLink next onClick={() => selectPage(currentPage + 1)} />
      </PaginationItem>
    </Pagination>
  );
};

export default PaginationComponent;
