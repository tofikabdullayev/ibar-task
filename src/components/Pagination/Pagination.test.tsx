import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Pagination from './index';

configure({ adapter: new Adapter() });
test('renders the component', () => {
  const component = shallow(
    <Pagination pageCount={1} currentPage={1} selectPage={() => {}} />
  );
  expect(component).toMatchSnapshot();
});
