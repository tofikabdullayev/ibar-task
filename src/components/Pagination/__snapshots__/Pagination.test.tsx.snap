// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Pagination
  aria-label="Page navigation"
  listTag="ul"
  style={
    Object {
      "flexWrap": "wrap",
      "marginTop": 20,
    }
  }
  tag="nav"
>
  <PaginationItem
    disabled={true}
    tag="li"
  >
    <PaginationLink
      onClick={[Function]}
      previous={true}
      tag="a"
    />
  </PaginationItem>
  <PaginationItem
    active={true}
    key="0_id"
    tag="li"
  >
    <PaginationLink
      onClick={[Function]}
      tag="a"
    >
      1
    </PaginationLink>
  </PaginationItem>
  <PaginationItem
    disabled={true}
    tag="li"
  >
    <PaginationLink
      next={true}
      onClick={[Function]}
      tag="a"
    />
  </PaginationItem>
</Pagination>
`;
